### PR TITLE
add spzeros(::Tuple{<:Integer})

### LINKED ISSUE #40505: `spzeros((2,))` is not defined
```julia
julia> using SparseArrays

julia> zeros((2,2))
2×2 Matrix{Float64}:
 0.0  0.0
 0.0  0.0

julia> zeros((2,))
2-element Vector{Float64}:
 0.0
 0.0

julia> spzeros((2,2))
2×2 SparseMatrixCSC{Float64, Int64} with 0 stored entries:
  ⋅    ⋅ 
  ⋅    ⋅ 

julia> spzeros((2,))
ERROR: MethodError: no method matching spzeros(::Tuple{Int64})
Closest candidates are:
  spzeros(::Type{Tv}, ::Type{Ti}, ::Tuple{Integer, Integer}) where {Tv, Ti} at /home/ab/src/julia/buildmaster/usr/share/julia/stdlib/v1.7/SparseArrays/src/sparsematrix.jl:1678
  spzeros(::Type{Tv}, ::Tuple{Integer, Integer}) where Tv at /home/ab/src/julia/buildmaster/usr/share/julia/stdlib/v1.7/SparseArrays/src/sparsematrix.jl:1681
  spzeros(::Tuple{Integer, Integer}) at /home/ab/src/julia/buildmaster/usr/share/julia/stdlib/v1.7/SparseArrays/src/sparsematrix.jl:1682
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[6]:1
```
can I prepare a PR?

### PR DESCRIPTION
**Pull request by abraunst**
Fixes #40505

### CODE REVIEW

**Comment by dkarrasch** [comment]
Shall we have a little test to make sure this "new functionality" doesn't regress?

**Comment by abraunst** [comment]
> Shall we have a little test to make sure this "new functionality" doesn't regress?

Sure

**Comment by abraunst** [comment]
I added two more missing versions of `spzeros` with type parameters and dims and added some tests.

File: stdlib/SparseArrays/test/sparsevector.jl [hunk 1]
```diff
@@ -75,7 +75,11 @@ end
 @testset "other constructors" begin
     # construct empty sparse vector
 
-    @test exact_equal(spzeros(Float64, 8), SparseVector(8, Int[], Float64[]))
+    for dims in (8, (8,))
+        @test exact_equal(spzeros(dims), SparseVector(8, Int[], Float64[]))
+        @test exact_equal(spzeros(Float64, dims), SparseVector(8, Int[], Float64[]))
+        @test exact_equal(spzeros(Float64, Int, dims), SparseVector(8, Int[], Float64[]))
```
Comment (dkarrasch) [inline stdlib/SparseArrays/test/sparsevector.jl:81]:
```suggestion
        @test exact_equal(spzeros(Float64, Int16, dims), SparseVector(8, Int16[], Float64[]))
```
Let's have some non-standard thing here to make sure we see the effect of the extra argument, and then merge.

**Review by dkarrasch — APPROVED** [review]

**Comment by abraunst** [comment]
There are two errors in the commit messages :sweat:  (`<` instead of `<:` and a wrong order in type parameters). Should I rebase + squash + rewrite the commit message? Otherwise please correct or simplify the message when merging.

**Comment by dkarrasch** [comment]
Typically, I delete all intermediate commit messages and only keep the PR title upon merging. So you still have a few minutes to update the title if you wish. 😉 But I don't think it's necessary IMHO.

**Comment by abraunst** [comment]
> Typically, I delete all intermediate commit messages and only keep the PR title upon merging. So you still have a few minutes to update the title if you wish.  But I don't think it's necessary IMHO.

Thanks! The title is fine :+1:

**Pull request merged**